home-manager/git.nix

```diff
@@ -26,9 +26,16 @@
     };
 
     # - They will be overridden by local hooks, currently I do nothing to merge global and local hooks
-    # - Remember to add executable permission for hooks files
+    #   If I want it, https://github.com/timokau/dotfiles/blob/dfee6670a42896cfb5a94fdedf96c9ed2fa1c9d2/home/git.nix#L3-L11 may be a good example
+    # - I don't have confident for executable permissions are reqiored or not for them, removing it worked. :<
     hooks = {
-      commit-msg = ../config/git/hooks/commit-msg.bash;
+      commit-msg = pkgs.writeShellScript "prevent_typo.bash" ''
+        #!/usr/bin/env bash
+
+        set -euo pipefail
+
+        ${lib.getBin pkgs.typos}/bin/typos --config "${config.xdg.configHome}/typos/_typos.toml" "$1"
+      '';
```

Comment:
Actual product

```
       │ File: /nix/store/3kag980k6bfid1xk2pvip4cqz888bzg2-git-hooks/commit-msg
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ #!/nix/store/5l50g7kzj7v0rdhshld1vx46rf2k5lf9-bash-5.2p26/bin/bash
   2   │ #!/usr/bin/env bash
   3   │
   4   │ set -euo pipefail
   5   │
   6   │ /nix/store/vqbm8lq5nr2dw983sbdr2hgfdchshixg-typos-1.16.25/bin/typos --config "/home/kachick/.config/typos/_ty
       │ pos.toml" "$1"
   7   │
```

Inserting needless line at last, but this writeShellScript style is used in many codes even in nixpkgs 🙆‍♂️

Reply:
```
   1   │ #!/nix/store/5l50g7kzj7v0rdhshld1vx46rf2k5lf9-bash-5.2p26/bin/bash
   2   │ #!/usr/bin/env bash
```

Double shebang... 😢

Reply:
https://github.com/kachick/dotfiles/commit/40e8df2e2a64c22ee1d5b1a0392376d08a0a9f3b